### PR TITLE
EASY-2115: return 404 when a requested bag does not exist

### PR DESCRIPTION
**Pull request by stephanHogenboom**
Fixes EASY-2115

#### When applied it will...
* 404 instead of 200 when a non-existing bag is requested

#### Related pull requests on github

repo                       | PR
-------------------------- | -----------------
easy-                      | [PR#](PRlink)

### CODE REVIEW

File: src/main/scala/nl.knaw.dans.easy.bagindex/server/BagIndexServletComponent.scala [hunk 1]
```diff
@@ -136,9 +136,8 @@ trait BagIndexServletComponent {
         .flatMap(uuid => databaseAccess.doTransaction(implicit c => database.getBagInfo(uuid)))
         .map(createResponse[BagInfo](bagInfo => <result>{toXml(bagInfo)}</result>)(bagInfo => "result" -> toJson(bagInfo)))
         .recoverWith {
-          case BagIdNotFoundException(_) => Success(createResponse[Unit](_ => <result/>)(_ => "result" -> JArray(List.empty))(()))
+          case BagIdNotFoundException(_) => Success(NotFound(createResponse[Unit](_ => <result/>)(_ => "result" -> JArray(List.empty))(())))
```
Comment (rvanheest) [inline src/main/scala/nl.knaw.dans.easy.bagindex/server/BagIndexServletComponent.scala]: I'd rather see a nice description telling the `bagId` doesn't exist, than an empty object in the body.

File: src/main/scala/nl.knaw.dans.easy.bagindex/server/BagIndexServletComponent.scala [hunk 2]
```diff
@@ -136,9 +136,8 @@ trait BagIndexServletComponent {
...
         }
-        .map(Ok(_))
```
Comment (rvanheest) [inline src/main/scala/nl.knaw.dans.easy.bagindex/server/BagIndexServletComponent.scala]: So where did this `Ok` go to? Shouldn't you add it to the `map` above?

Comment (stephanHogenboom) [reply src/main/scala/nl.knaw.dans.easy.bagindex/server/BagIndexServletComponent.scala]: It also returns a 200 if we don't explicitly map it to 200. I can move it above it you want. But i works fine like this for all tests (automatic and junit)

Comment (rvanheest) [reply src/main/scala/nl.knaw.dans.easy.bagindex/server/BagIndexServletComponent.scala]: yes please! I'd prefer explicitness over implicitness here!

File: src/main/scala/nl.knaw.dans.easy.bagindex/server/BagIndexServletComponent.scala [hunk 3]
```diff
@@ -136,9 +136,8 @@ trait BagIndexServletComponent {
         .flatMap(uuid => databaseAccess.doTransaction(implicit c => database.getBagInfo(uuid)))
         .map(createResponse[BagInfo](bagInfo => <result>{toXml(bagInfo)}</result>)(bagInfo => "result" -> toJson(bagInfo)))
         .recoverWith {
-          case BagIdNotFoundException(_) => Success(createResponse[Unit](_ => <result/>)(_ => "result" -> JArray(List.empty))(()))
+          case BagIdNotFoundException(_) => Success(NotFound(createResponse[Unit](_ => <result/>)(_ => s"bag with id $uuidStr could not be found")(())))
```
Comment (rvanheest) [inline src/main/scala/nl.knaw.dans.easy.bagindex/server/BagIndexServletComponent.scala]: Now you still have an empty `<result/>` object when requesting the XML response. I don't think you need the `createResponse` call here. The following should be enough:

```suggestion
          case BagIdNotFoundException(_) => Success(NotFound(s"bag with id $uuidStr could not be found"))
```

**Pull request merged**